### ISSUE TITLE
关于ANL的特殊设置疑惑

### ISSUE THREAD

**Issue by PunkMale** [Virusdoll/Active-Negative-Loss#1]
作者您好！
请问为什么只有ActiveNegativeLoss类中加入了`l1_norm = sum(p.abs().sum() for p in model.parameters())` ，而其他损失函数类都没有？

**Comment by xichenye0930** [comment]
你好，感谢你的提问！

为什么会加入 L1 正则化？
* 我们发现，直接在一些过于简单的数据集上使用 ANL 会导致模型出现过拟合的问题。
* 为了缓解过拟合，我们引入 L1 正则化。
* 相关的讨论与实验请参考论文 4.1 Empirical Understanding 的部分。

为什么其它损失函数不使用 L1 正则化？
* 对于其它方法，我们主要基于它们论文中的实验设置进行实验。
* 为了公平性，我们同样也实验了 APL 在 L1 正则化下的性能表现，其性能相较使用 L2 正则化并没有显著变化，故我们一直维持 L2 的设置。
* 相关的讨论与实验请参考论文附录中 D.1 Empirical Understanding 的部分。

**Comment by PunkMale** [comment]
但是我在其他损失函数里面没有看到L2正则化呀？

在计算损失函数的时候有一个`is_anl `参数：
`loss = loss_function(logits, labels, model) if self.is_anl else loss_function(logits, labels)`
只有使用ANL损失才会将模型传入损失函数添加正则化项，我的理解是这样。

**Comment by xichenye0930** [comment]
为什么其它损失函数没有 L2 正则化的相关代码？
* 使用 SGD 优化器的情况下，Weight Decay 和 L2 正则化基本上是等价的，故应用 Weight Decay 等同于应用 L2 正则化。
* 对所有非 ANL 的损失函数，在其配置文件中，`weight_decay` 字段总是不为 `0`，这保证它们都应用了 L2 正则化。

**Comment by PunkMale** [comment]
噢噢，原来是这样，怪我学艺不精，感谢大佬指点！
